docs/psoc6/quickref.rst

```diff
@@ -426,4 +426,47 @@ Methods
 ^^^^^^^
 All the methods(functions) given in :ref:`machine.SPI <machine.SPI>` class have been implemented in this port
 
+ADC (analog to digital conversion)
+----------------------------------
+
+On the PSoC6, a single ADC block with id - '0' is available. The ADC functionality is available on the
+following pins : "P10_0" - "P10_5".
+
+Use the :ref:`machine.ADC <machine.ADC>` class::
+
+    from machine import ADC, Pin
+
+    adc = ADC(Pin("P10_0"))        # create an ADC object on ADC pin
+    val = adc.read_u16()           # read a raw analog value in the range 0-65535
+    val = adc.read_uv()            # read an analog value in micro volts
+
+The PSoC6 port also supports :ref:`machine.ADC <machine.ADCBlock>` API to have control over the ADC configuration. Currently 
+PSoC6 supports only 1 12-bit SAR ADC with the following channel to pin mapping and the defaults are set accordingly:
```

Comment:
Because the 12 bits seem to be a parameter in the code, but must not be set to anything other than 12, you may want to make a note here stating this. Just to make it clear.